src/test/compile-fail/rfc-2126-crate-paths/crate-visibility-ambiguity.rs

```diff
@@ -0,0 +1,23 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![feature(crate_in_paths)]
+#![feature(crate_visibility_modifier)]
+
+mod m {
+    pub struct Z;
+    pub struct S1(crate (::m::Z)); // OK
+    pub struct S2(::crate ::m::Z); // OK
+    pub struct S3(crate ::m::Z); //~ ERROR `crate` can only be used in absolute paths
```

Comment:
I'm still a bit surprised that this doesn't work out with `crate` being a visibility modifier. What is the reason for it not to parse that way?

Reply:
I answered this earlier (https://github.com/rust-lang/rust/pull/45771#discussion_r150357994), but GitHub has hidden the message.